crates/rooch-framework/sources/address_mapping.move

```diff
@@ -46,10 +47,15 @@ module rooch_framework::address_mapping{
             let addr = table::borrow(&am.mapping, maddress);
             option::some(*addr)
         }else{
-            option::none<address>()
+            default_rooch_address(maddress)
         }
     }
     
+    fun default_rooch_address(maddress: MultiChainAddress): Option<address> {
+        let hash = blake2b256(&maddress.raw_address);
+        option::some(moveos_std::bcd::to_address(hash))
```

Comment:
I think creating a new address in the resolve function is not a good idea because we can not ensure that the address mapping exists via the resolve function.

Reply:
An available Rooch address must be returned here, otherwise MetaMask wallet users will prompt an error when they initiate a transaction for the first time: the multiaddress sender({}) mapping record is not exists.

The idea here is to temporarily use the ETH address to obtain the Rooch address through Hash, and after the transaction is executed, complete the creation of the Rooch account and the mapping of the address in post_execute.

The reason why the ETH public key is not used as the Rooch address as previously discussed is because the ETH public key cannot be obtained in the resolve contract method. If the resolve signature is reconstructed later, it should be changed to use the ETH public key as the Rooch address. Rooch address. But I feel that this solution is not universal enough. When we need to support a new L1 public chain, we need to adapt here.

Reply:
We can make this TBD. We can generate the address in the rust code. The rust code can get the ETH signature.

Reply:
https://github.com/rooch-network/rooch/issues/449